src/include/distributed/multi_progress.h

```diff
@@ -0,0 +1,43 @@
+/*-------------------------------------------------------------------------
+ *
+ * multi_progress.h
+ *    Declarations for public functions and variables used in progress
+ *    tracking functions in Citus.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef MULTI_PROGRESS_H
+#define MULTI_PROGRESS_H
+
+
+#include "fmgr.h"
+#include "nodes/pg_list.h"
+
+
+#if (PG_VERSION_NUM < 100000)
+/* Undefined on PostgreSQL <= 9.6 */
+#define DSM_HANDLE_INVALID 0
+extern Datum pg_stat_get_progress_info(PG_FUNCTION_ARGS);
+#endif
+
+typedef struct ProgressMonitorHeader
+{
+	uint64 processId;
+	int stepCount;
+} ProgressMonitorHeader;
```

Comment:
Could we call this `ProgressMonitorData` and include the steps? or alternatively `ProgressMonitorData` could contain a `ProgressMonitorHeader` and steps.